client/apps/Planning/PlanningList.tsx

```diff
@@ -185,7 +186,14 @@ export class PlanningListComponent extends React.PureComponent<IProps> {
             <React.Fragment>
                 <PlanningListSubNav />
                 <ListPanel
-                    groups={groups}
+                    groups={groups.filter((group) => {
+                        const groupDateToMoment = moment(group.date);
+                        const dateFilter = currentSearch.advancedSearch?.dates?.start;
```

Comment:
does this view only get rendered in search?

Reply:
Gets rendered in Combined, Planning only and Event only views

Reply:
Only in search mode? You're filtering out items that don't have some search filters. What if user isn't in search mode?

Reply:
If user isn't in search mode the filter would be pointless. Good point, I'll add a check beforehand so we don't iterate needlessly, we can pass the items right away

Reply:
![image](https://github.com/superdesk/superdesk-planning/assets/2076953/a80b4c0d-359a-47b0-ac45-329facebea02)

Needless iteration aside, wouldn't there have been a bug in previous code ^ where if you are not in search mode, there would be no groups visible? Since if `dateFilter` is `null` you were returning `false` from the filter function?

Reply:
Nope, date filter has a default of today, so in theory, it's never `null`

Reply:
I see. Still, no filter should mean no filtering should be happening. Which means all items should be returned. Even if all existing code paths return the same result before/after, your latest approach is still better.

There's a good [rule of least surprise](http://principles-wiki.net/principles:principle_of_least_surprise) that I haven't had a chance to mention to you yet that applies here.